app/src/main/java/org/wikipedia/page/PageTitle.kt

```diff
@@ -0,0 +1,212 @@
+package org.wikipedia.page
+
+import android.os.Parcelable
+import com.google.gson.annotations.SerializedName
+import kotlinx.parcelize.Parcelize
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.language.AppLanguageLookUpTable
+import org.wikipedia.page.Namespace.Companion.fromLegacyString
+import org.wikipedia.settings.SiteInfoClient.getMainPageForLang
+import org.wikipedia.staticdata.TalkAliasData
+import org.wikipedia.staticdata.UserTalkAliasData
+import org.wikipedia.util.StringUtil.addUnderscores
+import org.wikipedia.util.StringUtil.removeNamespace
+import org.wikipedia.util.StringUtil.removeUnderscores
+import org.wikipedia.util.UriUtil.decodeURL
+import org.wikipedia.util.UriUtil.encodeURL
+import java.util.*
+
+/**
+ * Represents certain vital information about a page, including the title, namespace,
+ * and fragment (section anchor target).  It can also contain a thumbnail URL for the
+ * page, and a short description retrieved from Wikidata.
+ *
+ * WARNING: This class is not immutable! Specifically, the thumbnail URL and the Wikidata
+ * description can be altered after construction. Therefore do NOT rely on all the fields
+ * of a PageTitle to remain constant for the lifetime of the object.
+ */
+@Parcelize
+class PageTitle(
+    private var _namespace: String?,
+    // TODO: remove this SerializedName when Tab list is no longer serialized to shared prefs.
+    @SerializedName("site") var wikiSite: WikiSite,
+      private var _text: String = "",
```

Comment:
Please remove the extra space at the beginning of the variable.

Reply:
done